### PR TITLE
Generalize repeat combinators

### PR DESCRIPTION
**Pull request by jdegoes**

### CODE REVIEW

**Comment by jdegoes** [comment]
@nequissimus any idea why this is failing?

**Comment by wi101** [comment]
`scalafmt`

**Comment by jdegoes** [comment]
@wi101 I tried that but it didn't seem to help...

File: core/shared/src/main/scala/scalaz/zio/IO.scala [hunk 1]
```diff
@@ -417,18 +417,42 @@ sealed abstract class IO[+E, +A] { self =>
...
+
+  /**
+   * Repeats this action with the specified schedule until the schedule
+   * completes, or until the first failure. In the event of failure the progress
+   * to date, together with the error, will be passed to the specified handler.
+   */
+  final def repeatOrElse0[E1 >: E, B, C](schedule: Schedule[A, B],
+                                         orElse: (E, B) => IO[E1, C]): IO[E1, Either[C, B]] = {
+    def loop(last: Option[() => B], state: schedule.State): IO[E1, Either[C, B]] =
+      self.redeem(
```
Comment (wi101) [inline core/shared/src/main/scala/scalaz/zio/IO.scala]: @jdegoes I think in scalafmt there is a problem with parentheses, I suggest if you change it `{` the `e => ` will be at the same line as redeem

File: project/ScalazBuild.scala [hunk 2]
```diff
@@ -58,10 +58,18 @@ object Scalaz {
...
+    resolvers += Resolver.sonatypeRepo("public"),
+    resolvers += Resolver.typesafeRepo("releases"),
+    resolvers += Resolver.typesafeIvyRepo("releases"),
+    resolvers += Resolver.sbtPluginRepo("releases"),
+    resolvers += Resolver.bintrayRepo("owner", "repo"),
+    resolvers += Resolver.jcenterRepo
   )
 
-  implicit class ModuleHelper(p: Project) {
+   implicit class ModuleHelper(p: Project) {
```
Comment (ashabhasa) [inline project/ScalazBuild.scala]: @jdegoes This line is not correctly formated, that's why the build in Travis fails.

**Comment by regiskuckaertz** [comment]
@jdegoes @NeQuissimus fyi, all our builds are failing at The Guardian because CI cannot download SBT. We've been in touch with the guys at Sonatype and it turns out they have an IP blacklisting algorithm that was too zealous. It seems this is affecting quite a lot of people.

**Comment by regiskuckaertz** [comment]
(they had to whitelist our range of IP addresses, I'm assuming they will do the same for Travis servers)

**Comment by wi101** [comment]
-_-" I thought Travis failed because of scalafmt, I didn't see UNRESOLVED DEPENDENCIES!
We have the same problem in https://github.com/scalaz/scalaz-ioqueue/pull/23

**Comment by regiskuckaertz** [comment]
It should be back now if you restart the build.

File: microsite/src/main/resources/microsite/data/menu.yml [hunk 3]
```diff
@@ -17,6 +17,8 @@ options:
       url: usage/bracket.html
     - title: Fibers
       url: usage/fibers.html
+    - title: Schedules
+      url/usage/schedules.html
```
Comment (NeQuissimus) [inline microsite/src/main/resources/microsite/data/menu.yml]: `url: usage/schedules.html`

**Pull request merged**